contracts/CBKLand.sol

```diff
@@ -0,0 +1,75 @@
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "./util.sol";
+
+contract CBKLand is Initializable, ERC721Upgradeable, AccessControlUpgradeable {
+
+    bytes32 public constant GAME_ADMIN = keccak256("GAME_ADMIN");
+    bytes32 public constant NO_OWNED_LIMIT = keccak256("NO_OWNED_LIMIT");
+
+    // Land specific
+    uint256 public constant LT = 0; // Land Tier
+    uint256 public constant LC = 1; // Land Chunk Id
+    uint256 public constant LX = 2; // Land Coordinate X
+    uint256 public constant LY = 3; // Land Coordinate Y
+
+    event LandMinted(address indexed minter, uint256 id, uint256 tier, uint256 chunkId);
+    event LandTransfered(address indexed from, address indexed to, uint256 id);
+
+    // TotalLand
+    uint256 landMinted;
+    // Avoiding structs for stats
+    mapping(uint256 => mapping(uint256 => uint256)) landData;
+
+    function initialize () public initializer {
+        __ERC721_init("CryptoBladesKingdoms Land", "CBKL");
+        __AccessControl_init_unchained();
+
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    modifier restricted() {
+        _restricted();
+        _;
+    }
+
+    function _restricted() internal view {
+        require(hasRole(GAME_ADMIN, msg.sender), "Not game admin");
+    }
+
+    function get(uint256 id) public view returns (uint256, uint256, uint256, uint256) {
+        return (landData[id][LT], landData[id][LC], landData[id][LX], landData[id][LY]);
+    }
+
+    // tier, chunkid, x, y
+    function getOwned(address owner) public view returns (uint256, uint256, uint256, uint256) {
+        uint256 id = tokenOfOwnerByIndex(owner, 0);
```

Comment:
i assume this is work in progress but in case it's not  I'd point out this kinda ssumes you will only ever own one land (maybe that's the intention? let me know)